python/pyspark/worker.py

```diff
@@ -550,7 +550,10 @@ def read_udtf(pickleSer, infile, eval_type):
 
     # See `PythonUDTFRunner.PythonUDFWriterThread.writeCommand'
     num_arg = read_int(infile)
-    arg_offsets = [read_int(infile) for _ in range(num_arg)]
+    offsets = [read_int(infile) for _ in range(num_arg)]
+    names = [utf8_deserializer.loads(infile) if read_bool(infile) else None for _ in range(num_arg)]
+    args_offsets = [offset for offset, name in zip(offsets, names) if name is None]
+    kwargs_offsets = {name: offset for offset, name in zip(offsets, names) if name is not None}
```

Comment:
here we zip the offsets and names and then filter the result twice.  Can we dedup it into one place?